src/validators/ClassValidator.ts

```diff
@@ -110,43 +111,45 @@ export class BsClassValidator {
     private verifyChildConstructor() {
         for (let key in this.classes) {
             let classStatement = this.classes[key];
-            let newMethod = classStatement.memberMap.new;
-            let ancestorNewMethod = this.getAncestorMember(classStatement, 'new');
+            const newMethod = classStatement.memberMap.new as ClassMethodStatement;
 
             if (
                 //this class has a "new method"
                 newMethod &&
                 //this class has a parent class
-                classStatement.parentClass &&
-                //this class's ancestors have a "new" method
-                ancestorNewMethod
+                classStatement.parentClass
             ) {
-                //verify there's a `super()` as the first statement in this member's "new" method
-                let firstStatement = ((newMethod as ClassMethodStatement).func?.body?.statements[0] as ExpressionStatement)?.expression as CallExpression;
+                //prevent use of `m.` anywhere before the `super()` call
+                const cancellationToken = new CancellationTokenSource();
+                let superCall: CallExpression;
+                newMethod.func.body.walk(createVisitor({
+                    VariableExpression: (expression, parent) => {
```

Comment:
Does this `VariableExpression` catch all of the situations where `m.` could be used?

Reply:
If you don't get a `VariableExpression` for `m` in `thing.m` then that should do it.